pytest -n 20 fails test_one_account_send_bcc_setting

I have installed `pytest-xdist` and run tests in parallel. When I run `pytest -n 6`, tests run faster. But with larger numbers they sometimes fail. With `-n 20` they fail reliably:
```
$ pytest -n 20 -x
...
FAILED tests/test_account.py::TestOnlineAccount::test_one_account_send_bcc_setting - assert 0 == 10
```

Here is the line where assertion is triggered:
```
    def test_one_account_send_bcc_setting(self, acfactory, lp):
        ac1 = acfactory.get_online_configuring_account()
        ac2_config = acfactory.peek_online_config()
        c2 = ac1.create_contact(email=ac2_config["addr"])
        chat = ac1.create_chat_by_contact(c2)
        assert chat.id > const.DC_CHAT_ID_LAST_SPECIAL
        wait_successful_IMAP_SMTP_connection(ac1)
        wait_configuration_progress(ac1, 1000)
    
        lp.sec("ac1: setting bcc_self=1")
        ac1.set_config("bcc_self", "1")
    
        lp.sec("send out message with bcc to ourselves")
        msg_out = chat.send_text("message2")
        ev = ac1._evlogger.get_matching("DC_EVENT_MSGS_CHANGED")
>       assert ev[2] == msg_out.id
E       assert 0 == 10
E         -0
E         +10

tests/test_account.py:465: AssertionError
```

The event I get is `('DC_EVENT_MSGS_CHANGED', 0, 0)`.

--- a/python/tests/conftest.py
+++ b/python/tests/conftest.py
@@ -194,11 +194,6 @@ def acfactory(pytestconfig, tmpdir, request, session_liveconfig, datadir):
             lib.dc_set_config(ac._dc_context, b"configured", b"1")
             return ac
 
-        def peek_online_config(self):
-            if not session_liveconfig:
-                pytest.skip("specify DCC_PY_LIVECONFIG or --liveconfig")
-            return session_liveconfig.get(self.live_count)
-
         def get_online_config(self, pre_generated_key=True):
             if not session_liveconfig:
                 pytest.skip("specify DCC_PY_LIVECONFIG or --liveconfig")

--- a/python/tests/test_account.py
+++ b/python/tests/test_account.py
@@ -447,32 +447,54 @@ class TestOnlineAccount:
         ac1.import_self_keys(dir.strpath)
 
     def test_one_account_send_bcc_setting(self, acfactory, lp):
-        ac1, ac2 = acfactory.get_two_online_accounts()
-        chat = self.get_chat(ac1, ac2)
+        ac1 = acfactory.get_online_configuring_account()
+        ac2 = acfactory.get_online_configuring_account()
 
-        lp.sec("ac1: setting bcc_self=1")
-        ac1.set_config("bcc_self", "1")
+        # Clone the first account: we will test if sent messages
+        # are copied to it via BCC.
+        ac1_clone = acfactory.clone_online_account(ac1)
+
+        wait_configuration_progress(ac1, 1000)
+        wait_configuration_progress(ac2, 1000)
+        wait_configuration_progress(ac1_clone, 1000)
+
+        chat = self.get_chat(ac1, ac2)
 
         self_addr = ac1.get_config("addr")
         other_addr = ac2.get_config("addr")
 
-        lp.sec("send out message with bcc to ourselves")
+        lp.sec("send out message without bcc to ourselves")
+        ac1.set_config("bcc_self", "0")
         msg_out = chat.send_text("message1")
-        # wait for send out (BCC)
-        assert ac1.get_config("bcc_self") == "1"
+        assert not msg_out.is_forwarded()
+
+        # wait for send out (no BCC)
         ev = ac1._evlogger.get_matching("DC_EVENT_SMTP_MESSAGE_SENT")
+        assert ac1.get_config("bcc_self") == "0"
+
+        # make sure we are not sending message to ourselves
+        assert self_addr not in ev[2]
+        assert other_addr in ev[2]
+        ev = ac1._evlogger.get_matching("DC_EVENT_DELETED_BLOB_FILE")
+
+        lp.sec("ac1: setting bcc_self=1")
+        ac1.set_config("bcc_self", "1")
+
+        lp.sec("send out message with bcc to ourselves")
+        msg_out = chat.send_text("message2")
+
+        # wait for send out (BCC)
+        ev = ac1._evlogger.get_matching("DC_EVENT_SMTP_MESSAGE_SENT")
+        assert ac1.get_config("bcc_self") == "1"
+
+        # now make sure we are sending message to ourselves too
         assert self_addr in ev[2]
         assert other_addr in ev[2]
         ev = ac1._evlogger.get_matching("DC_EVENT_DELETED_BLOB_FILE")
 
-        ac1._evlogger.consume_events()
-        lp.sec("send out message without bcc")
-        ac1.set_config("bcc_self", "0")
-        msg_out = chat.send_text("message2")
-        ev = ac1._evlogger.get_matching("DC_EVENT_SMTP_MESSAGE_SENT")
-        assert self_addr not in ev[2]
-        assert other_addr in ev[2]
-        ev = ac1._evlogger.get_matching("DC_EVENT_DELETED_BLOB_FILE")
+        # Second client receives only second message, but not the first
+        ev = ac1_clone._evlogger.get_matching("DC_EVENT_MSGS_CHANGED")
+        assert ac1_clone.get_message_by_id(ev[2]).text == msg_out.text
 
     def test_send_file_twice_unicode_filename_mangling(self, tmpdir, acfactory, lp):
         ac1, ac2 = acfactory.get_two_online_accounts()

--- a/python/tests/test_account.py
+++ b/python/tests/test_account.py
@@ -7,7 +7,6 @@ from deltachat import const, Account
 from deltachat.message import Message
 from datetime import datetime, timedelta
 from conftest import (wait_configuration_progress,
-                      wait_successful_IMAP_SMTP_connection,
                       wait_securejoin_inviter_progress)
 
 
@@ -448,37 +447,31 @@ class TestOnlineAccount:
         ac1.import_self_keys(dir.strpath)
 
     def test_one_account_send_bcc_setting(self, acfactory, lp):
-        ac1 = acfactory.get_online_configuring_account()
-        ac2_config = acfactory.peek_online_config()
-        c2 = ac1.create_contact(email=ac2_config["addr"])
-        chat = ac1.create_chat_by_contact(c2)
-        assert chat.id > const.DC_CHAT_ID_LAST_SPECIAL
-        wait_successful_IMAP_SMTP_connection(ac1)
-        wait_configuration_progress(ac1, 1000)
+        ac1, ac2 = acfactory.get_two_online_accounts()
+        chat = self.get_chat(ac1, ac2)
 
         lp.sec("ac1: setting bcc_self=1")
         ac1.set_config("bcc_self", "1")
 
+        self_addr = ac1.get_config("addr")
+        other_addr = ac2.get_config("addr")
+
         lp.sec("send out message with bcc to ourselves")
-        msg_out = chat.send_text("message2")
-        ev = ac1._evlogger.get_matching("DC_EVENT_MSGS_CHANGED")
-        assert ev[2] == msg_out.id
+        msg_out = chat.send_text("message1")
         # wait for send out (BCC)
         assert ac1.get_config("bcc_self") == "1"
-        self_addr = ac1.get_config("addr")
         ev = ac1._evlogger.get_matching("DC_EVENT_SMTP_MESSAGE_SENT")
         assert self_addr in ev[2]
+        assert other_addr in ev[2]
         ev = ac1._evlogger.get_matching("DC_EVENT_DELETED_BLOB_FILE")
 
         ac1._evlogger.consume_events()
         lp.sec("send out message without bcc")
         ac1.set_config("bcc_self", "0")
-        msg_out = chat.send_text("message3")
-        assert not msg_out.is_forwarded()
-        ev = ac1._evlogger.get_matching("DC_EVENT_MSGS_CHANGED")
-        assert ev[2] == msg_out.id
+        msg_out = chat.send_text("message2")
         ev = ac1._evlogger.get_matching("DC_EVENT_SMTP_MESSAGE_SENT")
         assert self_addr not in ev[2]
+        assert other_addr in ev[2]
         ev = ac1._evlogger.get_matching("DC_EVENT_DELETED_BLOB_FILE")
 
     def test_send_file_twice_unicode_filename_mangling(self, tmpdir, acfactory, lp):

